pandas/tests/plotting/test_frame.py

```diff
@@ -3244,6 +3248,67 @@ def test_plot_no_numeric_data(self):
         with pytest.raises(TypeError):
             df.plot()
 
+    @pytest.mark.slow
+    @pytest.mark.parametrize("method", ["bar", "barh"])
+    def test_bar_ticklabel_consistence(self, method):
+        # Draw two consecutiv bar plot with consistent ticklabels
+        # GH: 26186
+        def get_main_axis(ax):
+            if method == "barh":
+                return ax.yaxis
+            elif method == "bar":
+                return ax.xaxis
+
+        data = {"A": 0, "B": 3, "C": -4}
+        df = pd.DataFrame.from_dict(data, orient="index", columns=["Value"])
+        ax = getattr(df.plot, method)()
+        ax.get_figure().canvas.draw()
+        xticklabels = [t.get_text() for t in get_main_axis(ax).get_ticklabels()]
+        label_positions_1 = dict(zip(xticklabels, get_main_axis(ax).get_ticklocs()))
+        df = df.sort_values("Value") * -2
+        ax = getattr(df.plot, method)(ax=ax, color="red")
+        ax.get_figure().canvas.draw()
+        xticklabels = [t.get_text() for t in get_main_axis(ax).get_ticklabels()]
+        label_positions_2 = dict(zip(xticklabels, get_main_axis(ax).get_ticklocs()))
+        assert label_positions_1 == label_positions_2
+
+    def test_bar_numeric(self):
+        # Bar plot with numeric index have tick location values equal to index
+        # values
+        # GH: 11465
+        index = np.arange(10, 20, dtype=np.int64)
+        df = pd.DataFrame(np.random.rand(10), index=np.arange(10, 20))
+        ax = df.plot.bar()
+        ticklocs = ax.xaxis.get_ticklocs()
+        tm.assert_numpy_array_equal(ticklocs, index)
+
+    def test_bar_multiindex(self):
+        # Test from pandas/doc/source/user_guide/visualization.rst
+        # at section Plotting With Error Bars
+        # Related to issue GH: 26186
+
+        ix3 = pd.MultiIndex.from_arrays(
+            [
+                ["a", "a", "a", "a", "b", "b", "b", "b"],
+                ["foo", "foo", "bar", "bar", "foo", "foo", "bar", "bar"],
+            ],
+            names=["letter", "word"],
+        )
+
+        df3 = pd.DataFrame(
+            {"data1": [3, 2, 4, 3, 2, 4, 3, 2], "data2": [6, 5, 7, 5, 4, 5, 6, 5]},
+            index=ix3,
+        )
+
+        # Group by index labels and take the means and standard deviations
+        # for each group
+        gp3 = df3.groupby(level=("letter", "word"))
+        means = gp3.mean()
+        errors = gp3.std()
+
+        # No assertion we just ensure that we can plot a MultiIndex bar plot
+        means.plot.bar(yerr=errors, capsize=4)
```

Comment:
This test generally seems too loose; is this only because the MultiIndex doesn't render? If so again I *think* should raise

Reply:
The MultiIndex does render before the fix.
They were however no test for a bar plot of a multiindex.
I catched the error only through an example from the user guide.